packages/gatsby-theme-api-docs/src/hooks/use-type-locations.js

```diff
@@ -56,8 +57,15 @@ export const useTypeLocations = () => {
       }
     `
   );
+  const typeLocations = convertComponentInMdxToTypeLocations(queryResult);
+  const typeLocationOverrides = useTypeLocationOverrides().reduce(
+    (list, type) => {
+      return { ...list, [type.type]: { urlAnchorTag: type.path } };
```

Comment:
As discussed I am not happy with this data design.   The code that consumes the result of the hook will now have to have lots of null checks and special logic for type locations because overridden types will have some fields not set that types from the introspection hook would have set. 

In our review, we found that we can reduce the data structure to being just an href, not exposing the innards of the original GraphQL structures as much.